src/nerf/gun.gleam

```diff
@@ -11,11 +11,19 @@ pub external type StreamReference
 
 pub external type ConnectionPid
 
-pub fn open(host: String, port: Int) -> Result(ConnectionPid, Dynamic) {
-  open_erl(charlist.from_string(host), port)
+pub fn open(
+  host: String,
+  port: Int,
+  opts_map: Dynamic,
```

Comment:
This is the use of Dynamic I was referring to. Best swap it out for a precise type.

Perhaps we could also use the label `configuring` or such?

Maybe we'd want to also have the host and the port in the config? I'm not sure why those two would be passed independently.

Reply:
With updated types: 

```
import nerf/websocket.{Transport, Tls, TransportOpts, Verify, VerifyNone}

let opts = [Transport(Tls), TransportOpts([Verify(VerifyNone)])]
assert Ok(conn) = websocket.connect_with_options("example.com", "/ws", 443, [], opts)
```

Reply:
Fab!

It'll also need to be applied to this module too rather than it using Dynamic.

This permits transports being specified multiple times, is that valid? What does it do?

It looks like there's only one transport option of VerifyNone. Does it do something different when given to when not?

Is Verify specific to Tls? It can be given for Tcp currently.

Could we also document the options as well?

Thank you.

Reply:
These are the verify options, and `VerifyPeer` is default if nothing is set. 

```gleam
pub type TlsVerify {
  VerifyNone
  VerifyPeer
}```

It is only for TLS, having this setting separately is closer to way `gun` configures options in erlang. Having it as an array also is closer to what gun has, because allows to set only what you need to overwrite the defaults. It's also called `opts` in their docs, hence the `ConnectionOpts` name. Generally I was trying to stay close to the original lib and trying to re-design the shape of connect opts is somewhat out of scope here. Changing the shape of original settings and adding a converter to gun complicates adding new options as well. That said I've just pushed a change to: 

```
pub type ConnectionTransport {
  Tls(verify: TlsVerify)
  Tcp
}

ConnectionOpts(
  transport: ConnectionTransport
)
```